yaptide/batch/watcher.py

```diff
@@ -0,0 +1,90 @@
+from pathlib import Path
+import re
+import time
+import argparse
+import signal
+
+
+def log_generator(thefile):
+    """Generator function for monitoring purpose"""
+    while True:
+        line = thefile.readline()
+        if not line:
```

Comment:
I would add a timeout as well. If no new line is appended after some time then the generator will exit. This can be implemented using counter in the `while` loop. By default I would set the timeout to 1 hour (3600 seconds or 3600 passes of the loop). 

The timeout could be added as a parameter to the generator with default value o 3600

It may happen that for some reason the log file may stay permanently open for indefinite time and the generator will be stucked. This would cause the monitoring process to be hanging for ever.